src/Form/index.tsx

```diff
@@ -1065,13 +1066,27 @@ function Form({
       }
     } else {
       setFirst(false);
-      // @ts-expect-error TS(2531): Object is possibly 'null'.
       const nextStep = steps[redirectKey];
       const hasNext = nextStep.buttons.some((b: any) =>
         b.properties.actions.some((action: any) => action.type === ACTION_NEXT)
       );
       const terminalStep = !hasNext && nextStep.next_conditions.length === 0;
-      if (terminalStep) eventData.completed = true;
+      if (terminalStep) {
+        const authIntegration = getAuthIntegrationMetadata(integrations);
+        // Completed should be false if it is a 'terminal step' in the middle of
+        // a login flow with steps configured for after login
+        const isTerminalStepAuth =
+          authState.sentAuth &&
+          authIntegration?.auth_gate_steps.length &&
+          !authIntegration?.auth_gate_steps.includes(steps[stepKey].id);
+        // Need to guard against setting completed back to false actually this
+        // doesn't work properly, because if you create multiple accounts with
+        // the same fuser ID, the activation storm still thinks it's complete
+        // after logging out. so probably just need this to stay as true like we
+        // discussed (tested more and it actually can be set back to false)
+        // eventData.completed = formCompleted || !isTerminalStepAuth;
```

Comment:
Sorry, I realized I was wrong and actually I confirmed that I can overwrite the completed flag from true to false. 

I tried to guard against setting completed back to false, but actually it doesn't work properly. If you create multiple accounts with the same fuser ID, the activation form still thinks it's complete after logging out

so probably just need this to stay true like we discussed

Reply:
oh hmm whoops. i could pretty easily update the backend logic so completed can't be unset?

Reply:
that's at least the easiest solve from my end, so if you are fine with that change it sounds good to me. thinking about it more, I do think it makes sense to freeze completed once it's true - even if a user goes back and changes things they shouldn't be able to unset required fields